plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -1596,15 +1732,22 @@ public ObjectId getTransformationID( String name, RepositoryDirectoryInterface r
   public List<RepositoryElementMetaInterface> getTransformationObjects( ObjectId idDirectory, boolean includeDeleted )
     throws KettleException {
     return getPdiObjects( idDirectory, Arrays
-        .asList( new RepositoryObjectType[] { RepositoryObjectType.TRANSFORMATION } ), includeDeleted );
+      .asList( new RepositoryObjectType[] { RepositoryObjectType.TRANSFORMATION } ), includeDeleted );
   }
 
   protected List<RepositoryElementMetaInterface> getPdiObjects( ObjectId dirId, List<RepositoryObjectType> objectTypes,
                                                                 boolean includeDeleted ) throws KettleException {
     try {
 
       //RepositoryDirectoryInterface repDir = getRootDir().findDirectory( dirId );
-      RepositoryFile dirFile = pur.getFileById( dirId.getId() );
+      readWriteLock.readLock().lock();
+      RepositoryFile dirFile;
+      try {
+        dirFile = pur.getFileById( dirId.getId() );
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The read lock should not be release between this and lower in the same method. The lock should be kept.